src/app/frontend/deploy/deploy_controller.js

```diff
@@ -77,43 +57,22 @@ export default class DeployController {
     /** @private {!md.$dialog} */
     this.mdDialog_ = $mdDialog;
 
-    /** @private {boolean} */
-    this.isDeployInProgress_ = false;
+    /** @export {boolean}*/
+    this.isDeployInProgress = false;
+
+    /** @export {!Object}*/
+    this.publisher = {};
   }
 
   /**
-   * Deploys the application based on the sate of the controller.
+   * Notifies the child scopes to call their deploy methods.
    *
    * @export
    */
-  deploy() {
-    // TODO(bryk): Validate input data before sending to the server.
-
-    /** @type {!backendApi.AppDeploymentSpec} */
-    let appDeploymentSpec = {
-      containerImage: this.containerImage,
-      isExternal: this.isExternal,
-      name: this.name,
-      portMappings: this.portMappings.filter(this.isPortMappingEmpty_),
-      replicas: this.replicas,
-      namespace: this.namespace,
-    };
-
-    /** @type {!angular.Resource<!backendApi.AppDeploymentSpec>} */
-    let resource = this.resource_('/api/appdeployments');
-
-    this.isDeployInProgress_ = true;
-    resource.save(
-        appDeploymentSpec,
-        (savedConfig) => {
-          this.isDeployInProgress_ = false;
-          this.log_.info('Successfully deployed application: ', savedConfig);
-          this.state_.go(replicasetliststate);
-        },
-        (err) => {
-          this.isDeployInProgress_ = false;
-          this.log_.error('Error deploying application:', err);
-        });
+  deployBySelection() {
+    this.isDeployInProgress = true;
+    this.publisher.deploy().then(
+        ((deployCtrl) => ((status) => deployCtrl.isDeployInProgress = status))(this));
```

Comment:
This is equal to the code below.
```
this.publisher.deploy().then(
      (function(deployCtrl) {
        return function(status){
          deployCtrl.isDeployInProgress = status;
        };
      })(this));
```
It decorates the calllback function with the controller instance because it is modified in callback.
I may solve it in this way as well.
```
let deployCtrl = this;
this.publisher.deploy()
      .then(() => { deployCtrl.setDeployInProgress_(false); })
      .error(() => { deployCtrl.setDeployInProgress_(false); });
```